### ISSUE TITLE
[Enhancement]: Use WordPress password dashicons within WooCommerce sections

### ISSUE THREAD

**Issue by nielslange** [woothemes/woocommerce#53545]
### Describe the solution you'd like

Currently, there are three sections within WooCommerce where we display the password icon:

<table>
<tr>
<td valign="top" width="50%">My Account » Account details:
<br><br>

![Image](https://github.com/user-attachments/assets/3c10724b-99ad-4309-ad9a-2c1fd61d8331)

</td>
<td valign="top" width="50%">My Account » Login:
<br><br>

![Image](https://github.com/user-attachments/assets/3c66da22-c204-4e10-8ff3-6453665d219c)

</td>
</tr>
<tr>
<td colspan="2" valign="top">Orders:
<br><br>

<img width="1104" alt="Image" src="https://github.com/user-attachments/assets/f8812fb1-227e-4cef-987e-a7d221972f36" />

</td>
</tr>
</table>

Both the icon itself and the show/hide password states are different. The following screenshots show these differences:

<table>
<tr>
<td valign="top" width="50%">WordPress login / hide password:
<br>

Icon: [dashicons-visibility](https://developer.wordpress.org/resource/dashicons/#visibility)

![Image](https://github.com/user-attachments/assets/e514fdb7-b0ae-40d9-95a2-e49d8266cdbc)
</td>
<td valign="top" width="50%">WooCommerce login / hide password:
<br>

Icon: [WooCommerce font `\e010`](https://github.com/woocommerce/woocommerce/blob/96dd8a12d7193cce1b29c1af67dc88efcd4426e0/plugins/woocommerce/client/legacy/css/woocommerce-layout.scss#L422)

![Image](https://github.com/user-attachments/assets/72d53555-02b0-45e8-b5c1-689c40cc7750)
</td>
</tr>
<tr>
<td valign="top" width="50%">WordPress login / show password:
<br>

Icon: [dashicons-hidden](https://developer.wordpress.org/resource/dashicons/#hidden)

![Image](https://github.com/user-attachments/assets/391b8a01-3294-4f9d-9cd8-046a457d474b)
</td>
<td valign="top" width="50%">WooCommerce login / show password:
<br>

Icon: [WooCommerce font `\e010`](https://github.com/woocommerce/woocommerce/blob/96dd8a12d7193cce1b29c1af67dc88efcd4426e0/plugins/woocommerce/client/legacy/css/woocommerce-layout.scss#L422)

![Image](https://github.com/user-attachments/assets/bb5e451a-d119-41b3-9a06-3e1193ebb7a1)
</td>
</tr>
</table>

This inconsistency can cause confusion for users and complicates maintenance, as the WooCommerce icons are not as intuitive or aligned with the WordPress standards.

## Steps to reproduce

1. Go to `/wp-login.php` and toggle the password icon.
2. Go to `/wp-login.php` and toggle the password icon.
3. Go to `/my-account/edit-account/` and toggle the password icons.

## Propsal

I propose using the WordPress Dashicons for the WooCommerce password fields. The corresponding code can be found in https://github.com/woocommerce/woocommerce/blob/96dd8a12d7193cce1b29c1af67dc88efcd4426e0/plugins/woocommerce/client/legacy/css/woocommerce-layout.scss#L420-L428

### Changes:

- Use [dashicons-visibility](https://developer.wordpress.org/resource/dashicons/#visibility), for the My Account sections, when the password is hidden.
- Use [dashicons-hidden](https://developer.wordpress.org/resource/dashicons/#hidden), for the My Account sections, when the password is visible.
- Use [dashicons-visibility](https://developer.wordpress.org/resource/dashicons/#visibility), for previewing orders.

**Comment by wavvves** [comment]
Hi @nielslange, thanks for raising this.

Are there other examples where Dashicons have been intentionally preferred over WooCommerce-specific styles?

The current WooCommerce font icon has been in place for over five years now, and at a glance I spotted another usage in `.shipping-calculator-button::after`. Additionally, the My Account area is set to eventually transition to blocks, so the change can have a much shorter life span than what's in place right now :/

**Comment by nielslange** [comment]
> Are there other examples where Dashicons have been intentionally preferred over WooCommerce-specific styles?

Good question, @wavvves. I've only looked at the password icon, but haven't checked all individual icons we're using in the codebase.

> The current WooCommerce font icon has been in place for over five years now, and at a glance I spotted another usage in `.shipping-calculator-button::after.`

For `.shipping-calculator-button::after`, I see the icon `\e019`, while the password icon is `\e010`. That said, I do see that the password icon is also used on the order previous section on `/wp-admin/admin.php?page=wc-orders`:

<img width="1104" alt="Image" src="https://github.com/user-attachments/assets/3d523417-e275-4e95-b2a9-75c6f8c157d8" />

> Additionally, the My Account area is set to eventually transition to blocks, so the change can have a much shorter life span than what's in place right now :/

That's a valid point. I'd probably still update the icon. That way, when creating the new My Account area later, we'll automatically reuse the new icon.

**Comment by wavvves** [comment]
Yeah, since it's a rather small, safe, and quick change, let's add it to the backlog and see if gets squeezed in, or somebody picks it up with the good first issue tag 👍🏼

**Comment by elizaan36** [comment]
@nielslange Should there be a holistic transition from WooCommerce icons to dashicons? I'm not sure if that's been covered already, but an audit of all examples would help with consistency in the approach. Want to avoid users seeing one icon style in one place and not the other.

**Comment by nielslange** [comment]
> Should there be a holistic transition from WooCommerce icons to dashicons?

@elizaan36 I noticed that the title of this issue was misleading. With this issue, it was not my intention to replace all WooCommerce icons with WordPress dashicons, but to only replace the password icons. That said, do you want me to audit the use of WooCommerce icons to see which ones could potentially be replaced by WordPress icons? When creating this issue, my primary intension was that users see the same password icon across their site.

**Comment by elizaan36** [comment]
Ah ok, thanks for the clarification.

> audit the use of WooCommerce icons to see which ones could potentially be replaced by WordPress icons?

No I wouldn't say that's necessary right now, as long as the password icon is consistent across.

Makes sense to change the approach to use [the seen/unseen icons](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library) since Dashicons are [no longer being updated](https://make.wordpress.org/design/2020/04/20/next-steps-for-dashicons/). WDYT?

**Comment by nielslange** [comment]
> No I wouldn't say that's necessary right now, as long as the password icon is consistent across.

Yeah, that's what I had in mind as well.

> Makes sense to change the approach to use [the seen/unseen icons](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library) since Dashicons are [no longer being updated](https://make.wordpress.org/design/2020/04/20/next-steps-for-dashicons/). WDYT?

To display consistent password icons across all pages, I see the following two options:
1. We use the password Dashicons for all WooCommerce pages, despite that they are no longer updated, to ensure that the WooCommerce pages show the same password icons as the WordPress core pages.
2. We use the WP/GB password icon for all WooCommerce pages and create [a WordPress core ticket](https://core.trac.wordpress.org/newticket), to also use the WP/GB password icon for all WordPress core pages.

**Comment by elizaan36** [comment]
Oh my, this is a good example of where the upstream contribution pause will make an impact on the decision. I'd go with Option 1 in this case to create the consistency as you mentioned, then in the future we can align icons across WP and Woo to the Gutenberg set.
